tests/unit/test_adapter_yaml.py

```diff
@@ -582,3 +582,25 @@ def test_empty_list_inline_comment_indentation(self) -> None:
         result = fix_code(source, config)
 
         assert result == source
+
+    def test_underscore_for_decimal_integers(self) -> None:
+        """Make underscore for decimal integers configurable."""
+        source = dedent(
+            """\
+            small_ints: [1, 10, 100]
+            big_ints: [1000, 10000, 1000000, 100000000000000000000]
+            """
+        )
+        fixed_source = dedent(
+            """\
+            ---
+            small_ints: [1, 10, 100]
+            big_ints: [1_000, 10_000, 1_000_000, 100_000_000_000_000_000_000]
+            """  # noqa: E501
+        )
+        config = YamlfixConfig()
+        config.underscore_integer = True
+
+        result = fix_code(source, config)
+
+        assert result == fixed_source
```

Comment:
Just to make sure I'd add a test where the source has underscores, the config is set to `undersscore_integer = False` and make sure that the reverse procedure works well too

```python
        fixed_source = dedent(
            """\
            small_ints: [1, 10, 100]
            big_ints: [1000, 10000, 1000000, 100000000000000000000]
            """
        )
        source = dedent(
            """\
            ---
            small_ints: [1, 10, 100]
            big_ints: [1_000, 10_000, 1_000_000, 100_000_000_000_000_000_000]
            """  # noqa: E501
        )

        config = YamlfixConfig()
        result = fix_code(source, config)
        assert result == fixed_source